platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/text/TreeFormatter.java

```diff
@@ -44,6 +45,18 @@ protected void doAppend(String text) {
         this.current = new Node();
     }
 
+    /**
+     * By default, if a child node + the parent node have only a short amount of total text,
+     * the formatter will merge them both onto the same line.
+     * <p>
+     * If this is set to {@code true}, this behavior will be disabled.
+     *
+     * @param alwaysChildrenOnNewlines {@code true} = never merge nodes; {@code false} (default) = merge nodes with short total text
+     */
+    public void setAlwaysChildrenOnNewlines(boolean alwaysChildrenOnNewlines) {
```

Comment:
To me, this looks like it belongs as a parameter in the constructor instead of as mutable state of the class

Reply:
I thought it might make sense to toggle this for certain nodes within a single message but not for others.  No strong feelings though, I'll move it to the constructor until this is needed to keep this formatted simpler.